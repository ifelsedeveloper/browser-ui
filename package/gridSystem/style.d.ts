declare const StyledGrid: any;
declare const StyledColumn: any;
export { StyledGrid, StyledColumn };
