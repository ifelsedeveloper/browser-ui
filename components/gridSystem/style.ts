/* This Source Code Form is subject to the terms of the Mozilla Public
 * License. v. 2.0. If a copy of the MPL was not distributed with this file.
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

import styled, { css } from 'styled-components'
import { GridProps, ColumnProps } from './index'

const StyledGrid = styled.div`
  box-sizing: border-box;
  display: grid;
  grid-template-columns: repeat(${(p: GridProps) => p.columns ? p.columns : '12'}, 1fr);
  grid-gap: ${(p: GridProps) => p.gap ? p.gap : '15px'};
  padding: ${(p: GridProps) => p.padding ? p.padding : '0'};
  max-width: ${(p: GridProps) => p.width ? p.width : 'initial'};
  height: ${(p: GridProps) => p.height ? p.height : 'initial'};
  color: ${(p: GridProps) => p.textColor ? p.textColor : 'inherit'};
  background-color: ${(p: GridProps) => p.background ? p.background : 'inherit'};
  ${(p: GridProps) => p.backgroundImage ? 'background-image: ' + p.backgroundImage + ';' : ''}

  ${(p: GridProps) => p.disabled
    ? css`
      user-select: none;
      opacity: 0.25;
    ` : ''
  }
` as any

const StyledColumn = styled.div`
  box-sizing: border-box;
  position: relative;
  display: ${(p: ColumnProps) =>
    p.align || p.verticalAlign || p.direction ? 'flex' : 'block'
  };
  grid-column: ${(p: ColumnProps) => p.size ? `span ${p.size}` : 'span 12'};
  justify-content: ${(p: ColumnProps) => p.align ? p.align : 'initial'};
  align-items: ${(p: ColumnProps) => p.verticalAlign ? p.verticalAlign : 'initial'};
  background-color: ${(p: ColumnProps) => p.background ? p.background : 'inherit'};
  ${(p: GridProps) => p.backgroundImage ? 'background-image: ' + p.backgroundImage + ';' : ''}
  flex-direction: ${(p: ColumnProps) => p.direction ? p.direction : 'initial'};
` as any

export {
  StyledGrid,
  StyledColumn
}
