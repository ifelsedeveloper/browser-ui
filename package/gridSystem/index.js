"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const style_1 = require("./style");
class Grid extends React.PureComponent {
    render() {
        const { id, disabled, columns, theme, onClick, children } = this.props;
        return (React.createElement(style_1.StyledGrid, { id: id, disabled: disabled || false, theme: theme, onClick: onClick, columns: columns }, children));
    }
}
exports.Grid = Grid;
class Column extends React.PureComponent {
    render() {
        const { id, size, theme, children, onClick } = this.props;
        return (React.createElement(style_1.StyledColumn, { id: id, size: size, theme: theme, onClick: onClick }, children));
    }
}
exports.Column = Column;
//# sourceMappingURL=index.js.map