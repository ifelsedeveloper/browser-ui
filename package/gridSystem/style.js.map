{"version":3,"file":"style.js","sourceRoot":"","sources":["../../components/gridSystem/style.ts"],"names":[],"mappings":";;AAIA,yDAA+C;AAE/C,wCAAqC;AAErC,MAAM,UAAU,GAAG,2BAAM,CAAC,GAAG,CAAA;;;kCAGK,CAAC,CAAY,EAAE,EAAE,CAAC,CAAC,CAAC,OAAO,CAAC,CAAC,CAAC,CAAC,CAAC,OAAO,CAAC,CAAC,CAAC,IAAI;cAClE,CAAC,CAAY,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,SAAS,CAAC,IAAI,MAAM;aACzD,CAAC,CAAY,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,SAAS,CAAC,IAAI,GAAG;eACnD,CAAC,CAAY,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,UAAU,CAAC,IAAI,SAAS;YAC/D,CAAC,CAAY,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,QAAQ,CAAC,IAAI,SAAS;WAC3D,CAAC,CAAY,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,OAAO,CAAC,IAAI,SAAS;sBAC9C,CAAC,CAAY,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,iBAAiB,CAAC,IAAI,SAAS;sBACnE,CAAC,CAAY,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,iBAAiB,CAAC,IAAI,EAAE;;;;IAI9E,CAAC,CAAY,EAAE,EAAE,CAAC,CAAC,CAAC,QAAQ;IAC5B,CAAC,CAAC,uBAAG,CAAA;;;KAGJ,CAAC,CAAC,CAAC,EACN;CACM,CAAA;AAmBN,gCAAU;AAjBZ,MAAM,YAAY,GAAG,2BAAM,CAAC,GAAG,CAAA;;;;iBAId,CAAC,CAAc,EAAE,EAAE,CAAC,CAAC,CAAC,IAAI,CAAC,CAAC,CAAC,QAAQ,CAAC,CAAC,IAAI,EAAE,CAAC,CAAC,CAAC,SAAS;qBACrD,CAAC,CAAc,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,gBAAgB,CAAC,IAAI,SAAS;iBACxE,CAAC,CAAc,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,YAAY,CAAC,IAAI,SAAS;sBAC3D,CAAC,CAAc,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,iBAAiB,CAAC,IAAI,SAAS;sBACrE,CAAC,CAAc,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,iBAAiB,CAAC,IAAI,EAAE;oBAChE,CAAC,CAAc,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,eAAe,CAAC,IAAI,SAAS;cACzE,CAAC,CAAc,EAAE,EAAE,CAAC,kBAAQ,CAAC,CAAC,CAAC,KAAK,EAAE,UAAU,CAAC;;;;CAIvD,CAAA;AAIN,oCAAY"}