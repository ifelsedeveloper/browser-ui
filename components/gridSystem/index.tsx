/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

import * as React from 'react'
import { StyledGrid, StyledColumn } from './style'

export interface GridProps {
  id?: string,
  disabled?: boolean,
  children?: React.ReactNode,
  // Component styles
  columns?: string,
  padding?: string,
  gap?: string,
  width?: string,
  height?: string,
  textColor?: string,
  background?: string,
  backgroundImage?: string,
}

class Grid extends React.PureComponent<GridProps, {}> {
  render () {
    return (
      <StyledGrid
        id={this.props.id}
        disabled={this.props.disabled ? this.props.disabled : false}
        columns={this.props.columns}
        padding={this.props.padding}
        gap={this.props.gap}
        width={this.props.width}
        height={this.props.height}
        textColor={this.props.textColor}
        background={this.props.background}
        backgroundImage={this.props.backgroundImage}
      >
        {this.props.children}
      </StyledGrid>
    )
  }
}

export interface ColumnProps {
  id?: string,
  children?: React.ReactNode,
  // Component styles
  size?: number | string,
  align?: string,
  verticalAlign?: string,
  background?: string,
  backgroundImage?: string,
  direction?: string
}

class Column extends React.PureComponent<ColumnProps, {}> {
  render () {
    return (
      <StyledColumn
        id={this.props.id}
        size={this.props.size}
        align={this.props.align}
        verticalAlign={this.props.verticalAlign}
        background={this.props.background}
        backgroundImage={this.props.backgroundImage}
        direction={this.props.direction}
      >
        {this.props.children}
      </StyledColumn>
    )
  }
}

export { Grid, Column }
