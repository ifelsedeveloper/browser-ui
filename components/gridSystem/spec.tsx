/* global expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import { Grid, Column } from '../gridSystem'

describe('gridSystem tests', () => {
  describe('grid component', () => {
    it('matches the snapshot', () => {
      const component = <Grid />
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(<Grid id='grid' />)
      const assertion = wrapper.find('#grid').length
      expect(assertion).toBe(1)
    })

    it('defines a custom padding', () => {
      const wrapper = shallow(<Grid padding='20px' />)
      const assertion = wrapper.props().padding
      expect(assertion).toEqual('20px')
    })

    it('defines a custom gap', () => {
      const wrapper = shallow(<Grid gap='10px' />)
      const assertion = wrapper.props().gap
      expect(assertion).toEqual('10px')
    })

    it('defines a custom width', () => {
      const wrapper = shallow(<Grid width='13px' />)
      const assertion = wrapper.props().width
      expect(assertion).toEqual('13px')
    })

    it('defines a custom background', () => {
      const wrapper = shallow(<Grid background='cyan' />)
      const assertion = wrapper.props().background
      expect(assertion).toEqual('cyan')
    })
  })

  describe('column component', () => {
    it('matches the snapshot', () => {
      const component = <Column />
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(<Column id='column' />)
      const assertion = wrapper.find('#column').length
      expect(assertion).toBe(1)
    })

    it('defines a custom align', () => {
      const wrapper = shallow(<Column align='flex-start' />)
      const assertion = wrapper.props().align
      expect(assertion).toEqual('flex-start')
    })

    it('defines a custom verticalAlign', () => {
      const wrapper = shallow(<Column verticalAlign='flex-end' />)
      const assertion = wrapper.props().verticalAlign
      expect(assertion).toEqual('flex-end')
    })

    it('defines a custom background', () => {
      const wrapper = shallow(<Column background='purple' />)
      const assertion = wrapper.props().background
      expect(assertion).toEqual('purple')
    })
  })
})
