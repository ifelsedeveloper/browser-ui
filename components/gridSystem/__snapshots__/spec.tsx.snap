// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gridSystem tests column component matches the snapshot 1`] = `
<div
  className="sc-bwzfXH jVikWA"
  direction={undefined}
  id={undefined}
  size={undefined}
/>
`;

exports[`gridSystem tests grid component matches the snapshot 1`] = `
<div
  className="sc-bdVaJa dzQmNt"
  disabled={false}
  height={undefined}
  id={undefined}
  width={undefined}
/>
`;
